ballerina/websocket_service.bal

```diff
@@ -57,19 +57,17 @@ isolated service class WsService {
             return self.handleError(caller, wsText);
         }
 
-        WSPayload|json|error wsPayload = self.customHeaders != {}
-                                        ? wsText.cloneWithType(WSPayload) : value:fromJsonString(text);
-        if wsPayload is error {
-            return self.handleError(caller, wsPayload);
+        WSPayload|json|error payload = self.customHeaders != {}
+            ? wsText.cloneWithType(WSPayload) : value:fromJsonString(text);
+        if payload is error {
+            return self.handleError(caller, payload);
         }
-        string wsType = wsPayload is WSPayload ? <string>wsPayload.'type : DEFAULT_VALUE;
-        string connectionId = wsPayload is WSPayload && wsPayload?.id !is () ? <string>wsPayload?.id : DEFAULT_VALUE;
-
         if !self.customHeaders.hasKey(WS_SUB_PROTOCOL) {
-            return self.handleSubscriptionRequest(caller, connectionId, wsPayload);
+            return self.handleSubscriptionRequest(caller, DEFAULT_VALUE, payload);
```

Comment:
In this case, do we need the `connectionId`?

Reply:
No we could change the handleSubscriptionRequest's id parameter to have `string?` type

Reply:
Yeah, that'd be better.